### PR TITLE
Fetch Argo workflow logs

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Part of #1066.

Next step:
1. Support multiple steps.
1. Support long-running jobs.

### CODE REVIEW

File: pkg/workflow/workflow.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		"-o", "jsonpath={.items[0].metadata.name}")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("getWorkflowPodName error: %v\n%v", string(output), err)
+	}
+
+	return string(output), nil
+}
+
+func getPodLogs(podName string) (string, error) {
```
Comment (Yancey0623) [inline pkg/workflow/workflow.go:79]: `kubectl logs` would fetch all the logs, please add a TODO comment here to implement fetch logs by an incremental way.

File: pkg/sql/executor_ir.go [hunk 2]
```diff
@@ -163,6 +164,16 @@ func writeArgoFile(coulerFileName string) (string, error) {
 	return argoYaml.Name(), nil
 }
 
+func getWorkflowID(output string) (string, error) {
```
Comment (Yancey0623) [inline pkg/sql/executor_ir.go:167]: Duplicated with `workflow.go`#getWorkflowID ?

File: pkg/workflow/workflow.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	wf := reWorkflow.FindStringSubmatch(string(output))
+	if len(wf) != 2 {
+		return "", fmt.Errorf("parse workflow ID error: %v", output)
+	}
+
+	return wf[1], nil
+}
+
+func getWorkflowStatusPhase(job pb.Job) (string, error) {
+	cmd := exec.Command("kubectl", "get", "wf", job.Id, "-o", "jsonpath={.status.phase}")
```
Comment (typhoonzero) [inline pkg/workflow/workflow.go:58]: can we use https://github.com/kubernetes/client-go instead of calling the command line?

Comment (tonyyang-svail) [reply pkg/workflow/workflow.go:58]: What is the advantage of using the `client-go`? I am not sure if the `client-go` supports parsing the Argo workflow descriptions, so I may need to spend some time to test it out.

File: pkg/sql/executor_ir.go (hunk 2 above)
Comment (tonyyang-svail) [reply pkg/sql/executor_ir.go:167]: Yes, I will move workflow related code to `package workflow` in the next PR.

File: pkg/workflow/workflow.go (hunk 1 above)
Comment (tonyyang-svail) [reply pkg/workflow/workflow.go:79]: Sure, I will add this in the next PR.

**Review by Yancey0623 — APPROVED** [review]
LGTM

File: pkg/workflow/workflow.go (hunk 3 above)
Comment (Yancey0623) [reply pkg/workflow/workflow.go:58]: I created an issue: https://github.com/sql-machine-learning/sqlflow/issues/1362 to discuss go-client or kubectl, maybe we can move to make some disccusion.

**Pull request merged**